src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -230,12 +248,16 @@ private UpdateDocumentsCommand prepareRequestForAttachingDocumentToCase(CaseDocu
         return new UpdateDocumentsCommand(NULL_TTL, documentsList);
     }
 
+    private boolean isDocumentMovingCases(String documentCaseTypeId) {
+        return bulkScanExceptionRecordTypes.contains(documentCaseTypeId.trim());
+    }
+
     @Override
-    public String generateHashToken(UUID documentId) {
-        ResponseEntity<?> responseEntity = getDocumentMetadata(documentId);
+    public String generateHashToken(UUID documentId, Optional<StoredDocumentHalResource> documentMetaData) {
         String hashcodeFromStoredDocument = "";
-        if (responseEntity.getStatusCode().equals(HttpStatus.OK) && responseEntity.getBody() != null) {
-            StoredDocumentHalResource resource = (StoredDocumentHalResource) responseEntity.getBody();
+
+        if (documentMetaData.isPresent()) {
```

Comment:
callers shouldn't call this if there is no metadata. Change signature to String generateHashToken(UUID documentId, StoredDocumentHalResource documentMetaData)